src/geode/mesh/helpers/convert_solid_mesh.cpp

```diff
@@ -0,0 +1,99 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/mesh/helpers/convert_solid_mesh.h>
+
+#include <geode/mesh/builder/tetrahedral_solid_builder.h>
+#include <geode/mesh/core/tetrahedral_solid.h>
+
+namespace
+{
+    template < typename MeshFrom, typename Builder >
+    void copy_points( const MeshFrom& from, Builder& builder )
+    {
+        builder.create_vertices( from.nb_vertices() );
+        for( const auto v : geode::Range{ from.nb_vertices() } )
+        {
+            builder.set_point( v, from.point( v ) );
+        }
+    }
+
+    template < geode::index_t dimension >
+    bool all_polyhedra_are_simplex( const geode::SolidMesh< dimension >& solid )
+    {
+        for( const auto p : geode::Range{ solid.nb_polyhedra() } )
+        {
+            if( solid.nb_polyhedron_vertices( p ) > 4 )
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+} // namespace
+
+namespace geode
+{
+    template < index_t dimension >
+    absl::optional< std::unique_ptr< TetrahedralSolid< dimension > > >
+        convert_solid_mesh_into_tetrahedral_solid(
+            const SolidMesh< dimension >& solid )
+    {
+        if( !all_polyhedra_are_simplex( solid ) )
+        {
+            return absl::nullopt;
+        }
+        auto tet_solid = TetrahedralSolid< dimension >::create();
+        auto builder =
+            TetrahedralSolidBuilder< dimension >::create( *tet_solid );
+        copy_points( solid, *builder );
+        builder->reserve_tetrahedra( solid.nb_polyhedra() );
+        for( const auto p : Range{ solid.nb_polyhedra() } )
+        {
+            const std::array< index_t, 4 > vertices{ solid.polyhedron_vertex(
+                                                         { p, 0 } ),
+                solid.polyhedron_vertex( { p, 1 } ),
+                solid.polyhedron_vertex( { p, 2 } ),
+                solid.polyhedron_vertex( { p, 3 } ) };
+            builder->create_tetrahedron( vertices );
+        }
+        for( const auto p : Range{ solid.nb_polyhedra() } )
+        {
+            for( const auto e : Range{ 4 } )
+            {
+                if( const auto adjacent =
+                        solid.polyhedron_adjacent( { p, e } ) )
+                {
+                    builder->set_polyhedron_adjacent(
+                        { p, e }, adjacent.value() );
+                }
+            }
+        }
+        return std::move( tet_solid );
```

Comment:
```suggestion
        return tet_solid;
```
Could you try this? Should use NRVO instead of move.